File: tests/fixtures/QA-JV-102/must-not-fire/InStringTest.java
/**
 * Phase 1 fixture: patterns inside string literals must NOT fire.
 */
public class InStringTest {
    @Test
    public void documentsAntiPattern() {
        String warning = "Thread.sleep(2000) is an anti-pattern in tests";
        String example = "Avoid TimeUnit.SECONDS.sleep(5) — use WebDriverWait";
        assertNotNull(warning);
        assertNotNull(example);
    }
}

File: tests/fixtures/QA-PY-005/must-not-fire/in_string_test.py
"""Phase 1 fixture: patterns inside string literals must NOT fire."""


def test_documents_anti_pattern():
    doc = "time.sleep(5) is bad practice in tests"
    warning = 'Avoid time.sleep(10) — use polling instead'
    assert len(doc) > 0
    assert len(warning) > 0
